public/request/auth/reset-password.php

```diff
@@ -9,19 +11,17 @@
     'password' => 'required|confirmed|min:8|different:username',
 ]);
 
-$user = $input['username'];
 $passResetToken = $input['token'];
 $newPass = $input['password'];
 
-if (!isValidPasswordResetToken($user, $passResetToken)) {
+$targetUser = User::firstWhere('User', $input['username']);
+
+if (!$targetUser || !isValidPasswordResetToken($targetUser->username, $passResetToken)) {
     return back()->withErrors(__('legacy.error.token'));
 }
 
-changePassword($user, $newPass);
-
-// Perform auto-login:
-if (authenticateFromCookie($user, $permissions, $userDetails)) {
-    generateAppToken($user, $tokenInOut);
-}
+// Change the user's password and automatically log them in.
+changePassword($targetUser->username, $newPass);
+Auth::login($targetUser);
 
-return back()->with('success', __('legacy.success.password_change'));
+return redirect()->route('home')->with('success', __('legacy.success.password_change'));
```

Comment:
Really not sure what's going on, or if it's even worth investigating, but if you cause two validation errors when trying to change your password, a successful change will redirect to the login page instead of the home page.

1. Click link from reset password email
2. Click "Change Password" - get Password required error
3. Click "Change Password" - get Password required error
4. Enter a valid password twice. Click "Change Password" - get redirected to login page.

This also happens if you enter two passwords that don't match twice, or two blanks and two mismatches. It does not occur if you only generate a single validation error.

Reply:
I appreciate your thoroughness, as always. Weird, I can't seem to repro this issue. Wonder if it's env related? Tried a few different things but can't seem to get it to surface:

https://github.com/RetroAchievements/RAWeb/assets/3984985/1bbf19dd-2298-4295-851e-9708699ddbdd


https://github.com/RetroAchievements/RAWeb/assets/3984985/9d9206b5-342a-48f3-9fba-816753866315

Reply:
Seems to be working fine for me now too.